spec/facility_spec.rb

```diff
@@ -23,4 +25,186 @@
+
+      expect(@facility.registered_vehicles).to eq [@cruz, @bolt]
+    end
+
+    it 'can collect a fee when registering a vehicle' do
+      @facility.add_service('Vehicle Registration')
+      
+      @facility.register_vehicle(@cruz)
+      
+      expect(@facility.collected_fees).to eq 100
+    end
+
+    it 'can assigns the proper plate when registering a vehicle' do
+      @facility.add_service('Vehicle Registration')
+
+      @facility.register_vehicle(@cruz)
+      @facility.register_vehicle(@bolt)
+      @facility.register_vehicle(@camaro)
+
+      expect(@cruz.plate_type).to eq :regular
+      expect(@bolt.plate_type).to eq :ev
+      expect(@camaro.plate_type).to eq :antique
+    end
+  end 
+
+  describe '#collect_fee' do
+    before(:each) do
+      @facility.add_service('Vehicle Registration')
+      @cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice} )
+      @bolt = Vehicle.new({vin: '987654321abcdefgh', year: 2019, make: 'Chevrolet', model: 'Bolt', engine: :ev} )
+      @camaro = Vehicle.new({vin: '1a2b3c4d5e6f', year: 1969, make: 'Chevrolet', model: 'Camaro', engine: :ice} )
+    end
+
+    it 'can collect a fee (regular)' do
+      @facility.collect_fee(@cruz)
+
+      expect(@facility.collected_fees).to eq 100
+    end
+
+    it 'can collect a fee (ev)' do
+      @facility.collect_fee(@bolt)
+
+      expect(@facility.collected_fees).to eq 200
+    end
+    
+    it 'can collect a fee (antique)' do
+      @facility.collect_fee(@camaro)
+      
+      expect(@facility.collected_fees).to eq 25
+    end
+
+    it 'can accumulate collected_fees' do
+      @facility.collect_fee(@cruz)
+
+      expect(@facility.collected_fees).to eq 100
+
+      @facility.collect_fee(@bolt)
+
+      expect(@facility.collected_fees).to eq 300
+
+      @facility.collect_fee(@camaro)
+
+      expect(@facility.collected_fees).to eq 325
+    end
+  end 
+
+  describe '#administer_written_test' do
+    before(:each) do
+      @registrant_1 = Registrant.new('Bruce', 18, true)
+      @registrant_2 = Registrant.new('Penny', 16)
+      @registrant_3 = Registrant.new('Tucker', 15)
+    end
+
+    it 'cannot administer a written test if the service is not present at facility' do
+      expect(@facility.administer_written_test(@registrant_1)).to eq nil
+      expect(@registrant_1.license_data).to eq ({written: false, license: false, renewed: false})
+    end
+
+    it 'cannot administer a written test if the registrant does not have a permit' do
+      @facility.add_service('Written Test')
+      @facility.administer_written_test(@registrant_1)
+      @facility.administer_written_test(@registrant_2)
+
+      expect(@registrant_1.license_data).to eq ({:written=>true, :license=>false, :renewed=>false})
+      expect(@registrant_2.license_data).to eq ({:written=>false, :license=>false, :renewed=>false})
+    end
+
+    it 'cannot administer a written test if the registrant is younger than 16' do
```

Comment:
hmmm... you have tests for these things, but I didn't see the conditions handled in the methods.